src/esssans/beam_center_finder.py

```diff
@@ -252,13 +256,17 @@ def _cost(xy: List[float], *args) -> float:
     c = (all_q - ref) ** 2
     out = (sc.sum(ref * c) / sc.sum(ref)).value
     logger = get_logger('sans')
-    logger.info(f'Beam center finder: x={xy[0]}, y={xy[1]}, cost={out}')
     if not np.isfinite(out):
-        raise ValueError(
+        iofq_values = all_q.values
+        out = 1.0e6 * iofq_values[np.isfinite(iofq_values)].max()
+        logger.info(
             'Non-finite value computed in cost. This is likely due to a division by '
-            'zero. Try restricting your Q range, or increasing the size of your Q bins '
-            'to improve statistics in the denominator.'
+            'zero. The value was artificially changed to a large finite number. '
+            'If the results are not satisfactory, try restricting your Q range, or '
+            'increasing the size of your Q bins to improve statistics in the '
+            'denominator.'
         )
+    logger.info(f'Beam center finder: x={xy[0]}, y={xy[1]}, cost={out}')
     return out
```

Comment:
Is this an intentional double use of `out`? So this is returning something very different if it is not finite?

Reply:
If `out` is NaN, the optimizer will either fail or go round in circles and never finds a converged value.
So I'm returning a large cost instead of a NaN. I don't think I can modify the value anywhere else, because this function is used directly by the scipy wrapper.

Before, we used to raise as soon as the optimizer was trying to probe a region of space where there was almost no data. But actually, returning a large cost allows it to continue and look elsewhere.

Now that I think of it, it may work if I return `np.inf` instead of a large number. That would probably be better.